pkg/stanza/operator/helper/time.go

```diff
@@ -47,12 +47,13 @@ type TimeParser struct {
 
 // Unmarshal starting from default settings
 func (t *TimeParser) Unmarshal(component *confmap.Conf) error {
-	cfg := NewTimeParser()
-	err := component.Unmarshal(&cfg, confmap.WithIgnoreUnused())
+	err := component.Unmarshal(t, confmap.WithIgnoreUnused())
 	if err != nil {
 		return err
 	}
-	*t = cfg
+	if t.LayoutType == "" {
+		t.LayoutType = StrptimeKey
+	}
```

Comment:
Why do we need to set default values in the unmarshaller? If we need this one then why not others? Maybe this is just a matter of test cases creating the default config consistently?

Reply:
The code was doing this previously in the `NewTimeParser` path. Instead of creating a new struct, I just set the default here now.

Reply:
I don't understand why though. The pattern we use elsewhere is to instantiate a default struct and then unmarshal into it.

Reply:
no, that's the only instance this occurs afaict? By patching this code https://github.com/open-telemetry/opentelemetry-collector/pull/9750 works well.

Reply:
I think it is a well established pattern throughout the repo to create a default config and then unmarshal into them. For pkg/stanza operators, it is somewhat cryptic but is handled [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3ab6693be542cdc4d2eede7aabbada9a514298ae/pkg/stanza/operator/config.go#L109) where we've basically looked up a factory for the specific operator and asked it for a new (default) config. I don't think we should be handling default values here.

Reply:
I see, in that case, since the default value is set earlier, we can safely remove the check.

Reply:
Looks like that didn't work: the registry works well to set default values, but because `TimeParser` is optional on `helper.ParserConfig`, it is nil by default, and since it's not initialized with default values, they don't get applied later on.

Reply:
@atoulme Can't we do the trick from https://github.com/open-telemetry/opentelemetry-collector/blob/c72092e00151ce8e4cfef32c73b3a80d54076278/receiver/otlpreceiver/config.go#L68-L70 ? That is, set the configuration, but remove it during unmarshaling if the key is not set